invenio_userprofiles/forms.py

```diff
@@ -90,7 +90,8 @@ def validate_username(self, field):
 
         # Check if username is already taken.
         datastore = current_app.extensions["security"].datastore
-        user = datastore.find_user(username=field.data)
+        # The `_username` DB column stores a lower-case version of the display name.
+        user = datastore.find_user(_username=field.data.lower())
```

Comment:
What about using the [UserProfileProxy.get_by_username()](https://github.com/inveniosoftware/invenio-userprofiles/blob/master/invenio_userprofiles/models.py#L72)
```suggestion
        user = UserProfileProxy.get_by_username(field.data)
```

That way we can avoid replicate the inner function with access to the "private" `_username`.

Reply:
Comment: I think this is better fixed in Invenio-Accounts on the [model itself](https://github.com/inveniosoftware/invenio-accounts/blob/master/invenio_accounts/models.py#L188-L207) using the custom [comparator on the hybrid property](https://docs.sqlalchemy.org/en/14/orm/extensions/hybrid.html#building-custom-comparators). Something along the lines of:

```
from sqlalchemy.ext.hybrid import Comparator

@username.comparator
def username(cls):
     return CaseInsensitiveComparator(cls._username)
```

This ensures that anything using the model will perform a correct database query.

Reply:
Done, I implemented this in https://github.com/inveniosoftware/invenio-accounts/pull/476